prober/prober.go

```diff
@@ -77,6 +78,7 @@ type ProbeResult struct {
 func NewProber(opts ...ProberOption) *Prober {
 	p := &Prober{
 		timeout: 10 * time.Second,
+		probes:  make(map[string]Probe),
```

Comment:
nitpick: it is really rare we use `make`. Most of the time we use this construct:
```suggestion
		probes:  map[string]Probe{},
```